corehq/apps/case_importer/do_import.py

```diff
@@ -429,19 +442,21 @@ def _populate_updated_fields(config, row):
     return fields_to_update
 
 
-def _is_valid_id(uploaded_id, domain, cache):
+def _is_valid_id(uploaded_id, domain, cache, user_id, locations):
     if uploaded_id in cache:
         return cache[uploaded_id]
 
     owner = get_wrapped_owner(uploaded_id)
-    return _is_valid_owner(owner, domain)
+    return _is_valid_owner(owner, domain, user_id, locations)
 
 
-def _is_valid_owner(owner, domain):
+def _is_valid_owner(owner, domain, user_id=None, locations=ALL_LOCATIONS):
     return (
-        (isinstance(owner, CouchUser) and owner.is_member_of(domain)) or
-        (isinstance(owner, Group) and owner.case_sharing and owner.is_member_of(domain)) or
-        _is_valid_location_owner(owner, domain)
+        (
```

Comment:
nit: I would break this up into named conditionals so each piece is easier to decipher

Reply:
:+1: Good idea